### PR TITLE
Update .npmrc documentation

### PR DESCRIPTION
**Pull request by mdjasper**
I found that information about commenting lines in a `.npmrc` file was lacking from the official documentation. I've updated the doc to reflect currently supported ways to commenting lines in the npm config file as found through reading other sources and through experimentation with different ways of adding comments.

### CODE REVIEW

**Comment by iarna** [comment]
`.npmrc` files are parsed with https://github.com/npm/ini so `;` and `#` do explicitly mark [start-of-line comments](https://github.com/npm/ini/blob/master/ini.js#L76).

File: doc/files/npmrc.md [hunk 1]
```diff
@@ -41,6 +41,15 @@ sensitive credentials, they must be readable and writable _only_ by your user
 account (i.e. must have a mode of `0600`), otherwise they _will be ignored by
 npm!_
 
+#### Comments
+
+Because lines in an npm config file follow the `key = value` pattern, lines may be effectivly commmented-out by placing any character at the beginning of the line (creating a `key` which is not recoginized, or used). Common comment characters are `;`, `#`, or `//`.
```
Comment (iarna) [inline doc/files/npmrc.md]: `;` and `#` represent actual comments which won't end up in the key-space for loaded config, by contrast `//` does not and in fact can be a part of valid keys. For instance, your authentication key is stored as:

```
//registry.npmjs.org/:_authToken=AUTH-TOKEN-HERE
```

Comment (mdjasper) [inline doc/files/npmrc.md]: Great feedback. I reached the conclusion in the PR through experimentation. I will update the doc PR to reflect the ini config specified comments

**Comment by mdjasper** [comment]
Addressed and corrected an issue pointed out by @iarna on the initial PR on what constitutes valid comments in the `.npmrc` file

**Comment by legodude17** [comment]
Is this still in progress?

**Comment by mdjasper** [comment]
@legodude17 I believe it it complete, but happy for more review if anyone thinks that is necessary

**Comment by legodude17** [comment]
@mdjasper, then it LGTM.

Of course, I don't actually have any authority, but I think this is a good idea.

**Comment by zkat** [comment]
Thanks for the patch, @mdjasper! This was included as part of `npm@3.10.8`, so I'm gonna close the PR now. 🎉

**Pull request closed without merging**